src/System.Text.Json/tests/Serialization/DictionaryTests.cs

```diff
@@ -932,6 +933,102 @@ public static void DictionaryOfArrayOfDictionary()
             Assert.Equal(JsonString, json);
         }
 
+        private interface IClass { }
+
+        private class MyClass : IClass { }
+
+        private class MyFactory : JsonConverterFactory
+        {
+            public override bool CanConvert(Type typeToConvert)
+            {
+                return typeToConvert == typeof(IClass) || typeToConvert == typeof(MyClass);
+            }
+
+            public override JsonConverter CreateConverter(Type typeToConvert, JsonSerializerOptions options)
+            {
+                return new MyStuffConverter();
+            }
+        }
+
+        private class MyStuffConverter : JsonConverter<IClass>
+        {
+            public override IClass Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+            {
+                return new MyClass();
+            }
+
+            public override void Write(Utf8JsonWriter writer, IClass value, JsonSerializerOptions options)
+            {
+                writer.WriteNumberValue(1);
+            }
+        }
+
+        private static IEnumerable<(Type, string)> NestedDictionaryTypeData()
```

Comment:
Any comments here or in the test case to summarize what this is doing?  It looks to me like it expands out to a ton of different nesting combinations.  It looks mildly exponential in that each of the 4 outer iterations is more than doubling the size of baseDictionaryTypes.

Reply:
I added a comment in the source -

```
// This method generates 316 unique test cases for nested dictionaries up to 4
// levels deep, along with matching JSON, encompassing the various planes of
// dictionaries that can be combined: generic, non-generic, BCL, user-derived,
// immutable, mutable, readonly, concurrent, specialized.
```

Reply:
Are you ok with exponential behavior?  How many cases does it generate now?  What would happen if you change depth?

Reply:
It generates 316 test cases (run as 1 test) now at depth 4, and I don't think the depth should be increased beyond this. I think the exponential behavior is okay so long as it remains contained. Would comments discouraging changes to this method that increase the number of tests be okay?

I like this approach because it should hopefully make it easier to catch future changes that break nesting scenarios/permutations that one might not readily consider. If 316 is too much, would this kind of test be better as an outerloop run?

Reply:
Got it.  I’d just comment on the depth to caution on changes.  316 alone isn’t concerning: how much time does it take?

Reply:
Done! It takes ~0.35s, taking the total test duration from ~22.5s to ~22.85s.

Reply:
I see, seems reasonable to me.